pkg/common/configs/configvalidator.go

```diff
@@ -573,6 +573,19 @@ func checkLimit(limit Limit, existedUserName map[string]bool, existedGroupName m
 	return nil
 }
 
+func checkLimitsStructure(rootQueue *QueueConfig, partitionLimits []Limit) error {
+	if len(partitionLimits) > 0 && len(rootQueue.Limits) > 0 {
+		return fmt.Errorf("partition limits and root queue limits both exist")
+	}
+
+	// if root queue limits not defined, apply partition limits
+	if len(partitionLimits) > 0 && len(rootQueue.Limits) == 0 {
+		rootQueue.Limits = partitionLimits
```

Comment:
this solution is simple and it should work I think. However, this method is not idempotent as the `PartitionConfig` changed by this method will get failed if we pass `PartitionConfig` to this method again. It seems to me that is error-prone. In other words, the validator hate `PartitionConfig` has both partition-level limits and root-level limits, but it may produce such case itself.

Reply:
Thanks for the detailed review, indeed we should wipe out partition limits after applied them to root queue, already addressed that in the latest commit.